mhkit/dolfyn/io/nortek2_lib.py

```diff
@@ -242,16 +261,20 @@ def _check_index(idx, infile, fix_hw_ens=False):
             hwe[inds[(ib + 1) :]] += 1
             ens[inds[(ib + 1) :]] += 1
 
-    # This block fixes skips that originate from before this file.
-    delta = max(hwe[:N_id]) - hwe[:N_id]
-    for d, id in zip(delta, idx["ID"][:N_id]):
-        if d != 0:
-            FLAG = True
-            hwe[id == idx["ID"]] += d
-            ens[id == idx["ID"]] += d
+    if not dp:
+        # This block fixes skips that originate from before this file.
+        # Check first N id's and correct
+        delta = max(hwe[:N_id]) - hwe[:N_id]
+        for d, id in zip(delta, idx["ID"][:N_id]):
+            if d != 0:
+                FLAG = True
+                hwe[id == idx["ID"]] += d
+                ens[id == idx["ID"]] += d
+
+        if np.any(np.diff(ens) > 1) and FLAG:
+            idx["ens"] = np.unwrap(hwe.astype(np.int64), period=period) - hwe[0]
 
-    if np.any(np.diff(ens) > 1) and FLAG:
-        idx["ens"] = np.unwrap(hwe.astype(np.int64), period=period) - hwe[0]
+    return dp
```

Comment:
Its not clear to me what is going on in this function. Why are we doing all these calculations to return a boolean dp that is not modified? Is this modifying the idx data structures in place so that we do not need to return those values (if so shouldn't we be more explicit about what this function is doing)?

Reply:
I can try removing this bit of code and see if that breaks tests, because I find a little more redundant code each time I need to make changes to this reader. 
Unfortunately a lot of the RDI reader modifies data structures in place; if I had to guess the main reason this was done is to conserve memory for very large files (>1 Gb).

Reply:
A workaround for the in-place modification may be to add these data structures to the class (i.e. `self.idx`)

Reply:
Thanks for clarifying. 

I think we can both agree that modifying in place and not being explicit about it is confusing and not best practice. 

Long term I would like us to make a plan which cleans this up but short term let me know what you think is best which can be to leave as is and explicitly address in the future.

Reply:
Sounds good. In the interest of merging this PR, and knowing there is a lot of undocumented in-place modification in these readers, let's address it in a single PR in the future.

Reply:
Turns out that blip of code is unused.